src/services/fcmToken/sendChatNotification.js

```diff
@@ -0,0 +1,43 @@
+const {messaging} = require('firebase-admin');
+const UsersFunction = require('../../databases/functions/users');
+const FcmTokenFunction = require('../../databases/functions/fcmToken');
+const {User, FcmToken} = require('../../databases/models');
+
+const sendChatNotification = async (userTargetId, payload) => {
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId, {
+    shouldReturnObject: true
+  });
+
+  const userTargetToken = await FcmTokenFunction.findAllTokenByUserId(
+    FcmToken,
+    signedUser?.user_id
+  );
+
+  if (userTargetToken) {
+    try {
+      userTargetToken.forEach((user) => {
+        const newPayload = {
+          ...payload,
+          token: user?.token,
+          data: {
+            ...payload.data,
+            is_annoymous: signedUser?.is_user_anonymous?.toString()
+          }
+        };
+
+        messaging()
+          .send(newPayload)
+          .then((response) => {
+            console.log('response', response);
+          })
+          .catch((e) => {
+            console.log('error send chat notification in promise', user?.token, e);
+          });
+      });
+    } catch (error) {
+      console.log('error send chat notifications', error);
+    }
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code sends notifications to all tokens associated with a user. However, there are a few potential issues:

1. The `messaging().send(newPayload)` function returns a promise, but you're not waiting for it to resolve before moving on to the next token. This could potentially lead to unhandled promise rejections if an error occurs. Consider using `Promise.all()` to wait for all promises to resolve before proceeding.

2. The error handling within the `forEach` loop only logs the error and doesn't stop the execution of other promises. If one notification fails, the others will still be sent. Depending on your use case, you may want to stop sending further notifications if one fails.

3. Logging sensitive information like tokens can pose a security risk. Be careful about what information you log, especially in production environments.

Here's how you might address these issues:

```diff
-      userTargetToken.forEach((user) => {
-        const newPayload = {
-          ...payload,
-          token: user?.token,
-          data: {
-            ...payload.data,
-            is_annoymous: signedUser?.is_user_anonymous?.toString()
-          }
-        };
-
-        messaging()
-          .send(newPayload)
-          .then((response) => {
-            console.log('response', response);
-          })
-          .catch((e) => {
-            console.log('error send chat notification in promise', user?.token, e);
-          });
-      });
+      const promises = userTargetToken.map((user) => {
+        return new Promise((resolve, reject) => {
+          const newPayload = {
+            ...payload,
+            token: user?.token,
+            data: {
+              ...payload.data,
+              is_annoymous: signedUser?.is_user_anonymous?.toString()
+            }
+          };
+
+          messaging()
+            .send(newPayload)
+            .then(resolve)
+            .catch(reject);
+        });
+      });
+
+      try {
+        await Promise.all(promises);
+      } catch (error) {
+        console.error('Error sending chat notifications:', error);
+        throw error;
+      }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->